#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE_NAME = "squirrel_pointcloud_filter"

gen = ParameterGenerator()
gen.add("enabled", bool_t, 0, "", True)
gen.add("pointcloud_frame", str_t, 0, "", "depth_optical_frame")
gen.add("end_effector_frame", str_t, 0, "", "end_effector_link")
gen.add("use_const_end_effector_dim", bool_t, 0, "", True)
gen.add("end_effector_topic", str_t, 0, "", "end_effector_size")
gen.add("end_effector_radius", double_t, 0, "", 0.25, 0.0, 10.0)

exit(gen.generate(PACKAGE_NAME, "squirrel_pointcloud_filter", "EndEffectorFilter"))